main/src/library/RedBlackSet.flix

```diff
@@ -0,0 +1,296 @@
+///
+/// An experimental RedBlackSet type represented internally as a red-black tree.
+///
+///
+enum RedBlackSet[t] {
+    case RedBlackSet(RedBlackTree[t])
+}
+
+namespace RedBlackSet {
+
+    ///
+    /// Returns the size of `xs`.
+    ///
+    pub def size(xs: RedBlackSet[a]): Int32 =
+        let RedBlackSet(es) = xs;
+            RedBlackTree.size(es)
+
+    ///
+    /// Returns the empty set.
+    ///
+    /// `RedBlackSet#{}` is syntactic sugar for `empty` (`RedBlackSet#{} = empty()`).
+    ///
+    /// TODO: Investigate how to activate the RedBlackSet#{} syntax
```

Comment:
We won't have custom syntax like that. But since the plan is to **replace** our current implementations with yours, we will get the pretty syntactic sugar for free :)